ui/src/shared/components/ThresholdMarkers.tsx

```diff
@@ -50,41 +62,40 @@ const ThresholdMarkers: FunctionComponent<Props> = ({
         const onChangeMaxPos = ({y}) => handleDrag(index, 'maxValue', y)
         const onChangeMinPos = ({y}) => handleDrag(index, 'minValue', y)
 
-        if (threshold.lowerBound && threshold.upperBound) {
-          return (
-            <RangeThresholdMarkers
-              key={index}
-              yScale={yScale}
-              yDomain={yDomain}
-              threshold={threshold}
-              onChangeMinPos={onChangeMinPos}
-              onChangeMaxPos={onChangeMaxPos}
-            />
-          )
-        }
-
-        if (threshold.lowerBound) {
-          return (
-            <GreaterThresholdMarker
-              key={index}
-              yScale={yScale}
-              yDomain={yDomain}
-              threshold={threshold}
-              onChangePos={onChangePos}
-            />
-          )
-        }
-
-        if (threshold.upperBound) {
-          return (
-            <LessThresholdMarker
-              key={index}
-              yScale={yScale}
-              yDomain={yDomain}
-              threshold={threshold}
-              onChangePos={onChangePos}
-            />
-          )
+        switch (threshold.type) {
+          case 'greater':
+            return (
+              <GreaterThresholdMarker
+                key={index}
+                yScale={yScale}
+                yDomain={yDomain}
+                threshold={threshold}
+                onChangePos={onChangePos}
+              />
+            )
+          case 'lesser':
+            return (
+              <LessThresholdMarker
+                key={index}
+                yScale={yScale}
+                yDomain={yDomain}
+                threshold={threshold}
+                onChangePos={onChangePos}
+              />
+            )
+          case 'range':
+            return (
+              <RangeThresholdMarkers
+                key={index}
+                yScale={yScale}
+                yDomain={yDomain}
+                threshold={threshold}
+                onChangeMinPos={onChangeMinPos}
+                onChangeMaxPos={onChangeMaxPos}
+              />
+            )
+          default:
+            return null
```

Comment:
The user should never get here, right?  Perhaps throw an error here for bug tracking / development purposes?

Reply:
make it a compile time error!

```
      const neverThreshold: never = threshold

      throw new Error(`unexpected threshold type "${(neverThreshold as any).type}"`)
```
luv 2 typescript